flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -92,6 +94,15 @@ public void open(InputSplit ignored) throws IOException {
 		}
 	}
 
+	private Statement createStatement() throws SQLException {
+		if(resultSetType == 0 || resultSetConcurrency == 0) {
```

Comment:
this line might lead to confusion when a user only sets either the type or concurrency. 
IMO we should check with && here, require the user to set both type and concurrency at the same time, and throw a meaningful exception if only one these is 0.